### PR TITLE
Phantom types: Add transform

### PR DESCRIPTION
**Pull request by brianvanburken**
## Properties
- transform
- transformN

## Values
- perspective
- rotate
- rotate3d
- rotateX
- rotateY
- rotateZ
- scale
- scale2
- scale3d
- scaleX
- scaleY
- scaleZ
- skew
- skew2
- skewX
- skewY
- translate
- translate2
- translate3d
- translateX
- translateY
- translateZ

# Contributing
- [x] Each `Value` is an **open record** with a single field. The field's name is the value's name, and its type is `Supported`. For example `foo : Value { provides | foo : Supported }`
- [x] Each function returning `Style` accepts a **closed record** of `Supported` fields.
- [x] If a function returning `Style` takes a single `Value`, that `Value` should always support `inherit`, `initial`, and `unset` because all CSS properties support those three values! For example, `borderFoo : Value { foo : Supported, bar : Supported, inherit : Supported, initial : Supported, unset : Supported } -> Style`
- [x] If a function returning `Style` takes **more than one** `Value`, however, then **none** of its arguments should support `inherit`, `initial`, or `unset`, because these can never be used in conjunction with other values! For example, `border-radius: 5px 5px;` is valid CSS, `border-radius: inherit;` is valid CSS, but `border-radius: 5px inherit;` is invalid CSS. To reflect this, `borderRadius : Value { ... } -> Style` must have `inherit : Supported` in its record, but `borderRadius2 : Value { ... } -> Value { ... } -> Style` must **not** have `inherit : Supported` in *either* argument's record. If a user wants to get `border-radius: inherit`, they must call `borderRadius`, not `borderRadius2`! 
- [x] When accepting a numeric `Value` (e.g. a length like `px`, an angle like `deg`, or a unitless number like `int` or `num`), *always* include `zero : Supported` as well as `calc : Supported`!
- [x] Every exposed value has documentation which includes at least 1 code sample.
- [x] Documentation links to to a [**CSS Tricks** article](https://css-tricks.com/) if available, and [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS) if not.
- [x] Make a pull request against the `phantom-types` branch, not `master`!

### CODE REVIEW

File: src/Css.elm [hunk 1]
```diff
@@ -11417,3 +11478,740 @@ columnRule3 :
...
+transform (Value val) =
+    AppendProperty ("transform:" ++ val)
+
+
+{-| Sets [`transform`](https://css-tricks.com/almanac/properties/t/transform/).
+
+    transformN [ (translate (px 12)), (scale 2), (skew (deg 20)) ]
+
+-}
+transformN :
```
Comment (brianvanburken) [inline src/Css.elm]: @rtfeldman do you know a way to enforce one or more in a list? This functions allows for zero or more right now

Comment (stoeffel) [reply src/Css.elm]: @brianvanburken you can use a tuple for that `(a, List a)`.

Comment (rtfeldman) [reply src/Css.elm]: `( a, List a )` works, although I think stylistically elsewhere in `elm-css` we've used `a -> List a -> ...` (as in, first argument is the first element in the list, second argument is the rest of it) so we should probably stick with that to be consistent!

File: src/Css.elm [hunk 2]
```diff
@@ -4199,7 +4260,7 @@ fontVariantNumeric4 val1 val2 val3 val4 =
 
 maybeValToString : Maybe (Value a) -> Maybe String
 maybeValToString =
-    Maybe.map (\(Value val) -> val)
+    Maybe.map unpackValue
```
Comment (rtfeldman) [inline src/Css.elm]: 👍 makes sense!

File: src/Css.elm (hunk 1 above)
Comment (rtfeldman) [reply src/Css.elm]: That said, @brianvanburken in cases where there's a useful and obvious value for `[]` to output, I like to do that and mention it in the docs.

So in this case I'd add something to the documentation about "if you pass `[]` you'll get `transform: none`" and implement that behavior. I think that'll be more useful than requiring a minimum of 1 element.

Comment (rtfeldman) [reply src/Css.elm]: One other note on this one: I think it's fine to rename `transformN` to `transform` and have that be the only option.

That will make it easier to discover, and adding an extra `[` and `]` around the argument shouldn't be too much bother. 😄

**Comment by rtfeldman** [comment]
Thanks @brianvanburken! Looks awesome! 😻

**Pull request merged**